### PR TITLE
Add unmark procedures for roles and tables + tests

### LINKED ISSUE #333: Add unlabel versions for relation, aid col, role
Continuation of #327. I think these should be separate functions to avoid potential ambiguities.
Something like `unmark_table`, `unmark_column`, `unmark_role`.

### PR DESCRIPTION
**Pull request by pdobacz**
Closes #333 

Not 100% sure I've got all the conclusions of the discussion on the subject, while I was away, but there it is. If the naming or interface is off, lmk

### CODE REVIEW

**Comment by pdobacz** [comment]
build fails on 14, hold off the review pls :/. Testing will require some more effort

**Comment by pdobacz** [comment]
I decided to fix the testing discrepancy between 14 and 13 by `ORDERING BY` in the `show_labels`. I think this is a nice feature regardless, to have the results of the function sorted in some sane order, and stable.

File: test/expected/_setup.out [hunk 1]
```diff
@@ -42,3 +42,61 @@ GRANT CONNECT ON DATABASE contrib_regression TO diffix_test;
 GRANT SELECT ON ALL TABLES IN SCHEMA public TO diffix_test;
 -- Allow SELECT on future test-specific tables.
 ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT ON TABLES TO diffix_test;
+-- Mark, unmark objects, show their labels
```
Comment (edongashi) [inline test/expected/_setup.out:45]: Not really relevant to setup. I would move these tests to `admin` or `misc`.

File: pg_diffix--0.0.1.sql [hunk 2]
```diff
@@ -34,7 +34,8 @@ LANGUAGE SQL
 AS $$
   SELECT objtype, objname, label
   FROM pg_seclabels
-  WHERE provider = 'pg_diffix';
+  WHERE provider = 'pg_diffix'
+  ORDER BY objtype, objname;
```
Comment (edongashi) [inline pg_diffix--0.0.1.sql:38]: Not necessary to do this for the sake of testing. We can filter some known subset to get a deterministic order. See `validation.sql` which filters them.

Comment (edongashi) [reply pg_diffix--0.0.1.sql:38]: > And does filtering itself guarantee stable order?

If the tests run then it's fine. The result order is possibly related to insertion order, and we always mark tables before columns. Ordering is fine but it's not necessary in my opinion.

File: pg_diffix--0.0.1.sql [hunk 3]
```diff
@@ -62,6 +63,15 @@ AS $$
   END;
 $$ LANGUAGE plpgsql;
 
+CREATE PROCEDURE unmark_table(table_name text)
+AS $$
+  BEGIN
+    DELETE FROM pg_catalog.pg_seclabel WHERE provider = 'pg_diffix' AND objoid = table_name::regclass::oid AND label = 'aid';
```
Comment (edongashi) [inline pg_diffix--0.0.1.sql:69]: Can't we do the same for table?

File: test/expected/validation.out [hunk 4]
```diff
@@ -250,8 +250,8 @@ SELECT * FROM diffix.show_settings() LIMIT 2;
 SELECT * FROM diffix.show_labels() WHERE objname LIKE 'public.test_customers%';
  objtype |         objname          |  label   
 ---------+--------------------------+----------
- table   | public.test_customers    | personal
  column  | public.test_customers.id | aid
+ table   | public.test_customers    | personal
```
Comment (edongashi) [inline test/expected/validation.out:254]: I would prefer tables be shown first. You usually care more about knowing which table is personal.

File: pg_diffix--0.0.1.sql (hunk 3 above)
Comment (pdobacz) [reply pg_diffix--0.0.1.sql:69]: I don't know what you mean? These lines are done for a table

Comment (edongashi) [reply pg_diffix--0.0.1.sql:69]: For table there is `SECURITY LABEL FOR pg_diffix ON TABLE ' || table_name || ' IS NULL`.
I think the same `DELETE` can clean up both table and AID cols.

Comment (pdobacz) [reply pg_diffix--0.0.1.sql:69]: Ah, I see, remove the `label = 'aid'` and then we don't need the setting to NULL. Yep this works, isn't this a bit too "magic" though? And not 100% sure if there cannot be something else sitting with the same `objoid`, other than columns :thinking:

Comment (pdobacz) [reply pg_diffix--0.0.1.sql:69]: but, on second thought, we unmark the table, so whatever is tagged onto it and concerning `pg_diffix` is fine to be unlabelled. I'll proceed like you suggest :+1:

Comment (edongashi) [reply pg_diffix--0.0.1.sql]: > but, on second thought, we unmark the table, so whatever is tagged onto it and concerning `pg_diffix` is fine to be unlabelled. I'll proceed like you suggest +1

Yes, this was my idea. Clear anything related to that table.

**Review by edongashi — DISMISSED** [review]

File: test/sql/admin.sql [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+-- Mark, unmark objects, show their labels
+CALL diffix.mark_role('diffix_test', 'direct');
+SELECT * FROM diffix.show_labels();
+CALL diffix.unmark_table('public.test_products');
+CALL diffix.unmark_role('diffix_test');
+SELECT * FROM diffix.show_labels();
+CALL diffix.mark_role('diffix_test', 'anonymized_trusted');
+CALL diffix.mark_public('public.test_products');
+SELECT * FROM diffix.show_labels();
```
Comment (edongashi) [inline test/sql/admin.sql:11]: Good for having this file around. `validation.sql` was getting bloated. It still is I think.

**Review by edongashi — APPROVED** [review]

**Pull request merged**